src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -399,32 +491,50 @@ def open_resolve_dialog(self, slack_request: SlackActionRequest, group: Group) -
-
         # XXX(CEO): the second you make a selection (without hitting Submit) it sends a slightly different request
-        modal_payload = self.build_resolve_modal_payload(callback_id)
-        try:
-            payload = {
-                "view": orjson.dumps(modal_payload).decode(),
-                "trigger_id": slack_request.data["trigger_id"],
-            }
-            headers = {"content-type": "application/json; charset=utf-8"}
-            slack_client.post(
-                "/views.open",
-                data=orjson.dumps(payload).decode(),
-                headers=headers,
-            )
-        except ApiError as e:
-            logger.exception(
-                "slack.action.response-error",
-                extra={
-                    "error": str(e),
-                    "organization_id": org.id,
-                    "integration_id": slack_request.integration.id,
+        if features.has("organizations:slack-sdk-action-view-open", group.project.organization):
+            modal_payload = self.build_resolve_modal_payload(callback_id)
+            slack_client = SlackSdkClient(integration_id=slack_request.integration.id)
+            try:
+                slack_client.views_open(
+                    trigger_id=slack_request.data["trigger_id"],
+                    view=modal_payload,
+                )
+            except SlackApiError:
+                logger.exception(
+                    "slack.action.response-error",
+                    extra={
+                        "organization_id": org.id,
+                        "integration_id": slack_request.integration.id,
+                        "trigger_id": slack_request.data["trigger_id"],
+                        "dialog": "resolve",
+                    },
+                )
+        else:
+            modal_payload = self.build_resolve_modal_payload_deprecated(callback_id)
+            slack_client = SlackClient(integration_id=slack_request.integration.id)
+            try:
+                payload = {
+                    "view": orjson.dumps(modal_payload).decode(),
                     "trigger_id": slack_request.data["trigger_id"],
-                    "dialog": "resolve",
-                },
-            )
+                }
+                headers = {"content-type": "application/json; charset=utf-8"}
+                slack_client.post(
+                    "/views.open",
+                    data=orjson.dumps(payload).decode(),
+                    headers=headers,
+                )
+            except ApiError as e:
+                logger.exception(
+                    "slack.action.response-error",
+                    extra={
+                        "error": str(e),
+                        "organization_id": org.id,
+                        "integration_id": slack_request.integration.id,
+                        "trigger_id": slack_request.data["trigger_id"],
+                        "dialog": "resolve",
+                    },
```

Comment:
We should reduce some of the duplication between the two `except` blocks if we can. The common elements of `extra` could be extracted into a local variable before entering the if-else block (I think we're doing this elsewhere with variables named `logger_params`).

If we want to add an extra `"error"` value in the `ApiError` case, an easy way to write it is `dict(error=str(e), **logger_params)`.

Also, is there a particular reason we're capturing `str(e)` here but not from the `SlackApiError`? Does the `SlackApiError` not come with a meaningful message? Or are they too verbose? (If it's that thing where they include an entire stack trace jammed into the message field, maybe we could trim from the first line break like we're doing elsewhere. And also maybe create a centralized utility for that sort of thing, if we're doing it a lot.)

Reply:
This logic will eventually be deleted so I am not really changing much here.

Reply:
logger.exception should automatically grab the exception for us.

Reply:
Okay.